packages/twenty-server/src/engine/core-modules/health/indicators/app.health.ts

```diff
@@ -14,57 +17,39 @@ export class AppHealthIndicator {
 
   constructor(
     private readonly healthIndicatorService: HealthIndicatorService,
-    private readonly objectMetadataService: ObjectMetadataService,
+    @InjectRepository(Workspace, 'core')
+    private readonly workspaceRepository: Repository<Workspace>,
     private readonly workspaceMigrationService: WorkspaceMigrationService,
   ) {}
 
   async isHealthy(): Promise<HealthIndicatorResult> {
     const indicator = this.healthIndicatorService.check('app');
 
     try {
-      const workspaces = await this.objectMetadataService.findMany();
-      const workspaceIds = [...new Set(workspaces.map((w) => w.workspaceId))];
-
-      const workspaceStats = await Promise.all(
-        workspaceIds.map(async (workspaceId) => {
-          const pendingMigrations =
-            await this.workspaceMigrationService.getPendingMigrations(
-              workspaceId,
-            );
+      const criticalWorkspaces =
```

Comment:
"criticalWorkspace" is a new naming that does not make sense to me.
proposal: "erroredWorkspace"